### PR TITLE
BAAS-14809: check for APIKeys as inputs to cli login command before opening browser

### PR DESCRIPTION
**Pull request by jandersongo**

### CODE REVIEW

File: internal/commands/login/inputs_test.go [hunk 1]
```diff
@@ -194,6 +193,17 @@ func TestLoginInputs(t *testing.T) {
 			test:        func(t *testing.T, i inputs) {},
 			expectedURL: apiKeysPage,
 		},
+		{
+			description: "should not open browser when api keys are provided as inputs",
```
Comment (makesitgo) [inline internal/commands/login/inputs_test.go:197]: [meganit] probably worth duplicating this for the username/password flags

File: internal/commands/login/inputs_test.go [hunk 2]
```diff
@@ -229,27 +239,39 @@ func TestLoginInputs(t *testing.T) {
...
+		i := inputs{AuthType: authTypeLocal}
 
 		profile := mock.NewProfile(t)
 		assert.Nil(t, i.Resolve(profile, ui))
 
-		assert.Equal(t, "there was an issue opening your browser\n", out.String())
+		console.Tty().Close() // flush the writers
+		<-doneCh              // wait for procedure to complete
+
+		assert.True(t, strings.HasPrefix(out.String(), "there was an issue opening your browser"), "should show error")
```
Comment (makesitgo) [inline internal/commands/login/inputs_test.go:274]: out of curiosity, what's the whole `out.String()` output?  wondering if we can do better/stricter than just a `HasPrefix` check

**Review by makesitgo — APPROVED** [review]
looks good to me... just one question about test assertion

🆒  beans

File: internal/commands/login/inputs_test.go (hunk 2 above)
Comment (jandersongo) [reply internal/commands/login/inputs_test.go:274]: the mock console prompts for the username/password
`"there was an issue opening your browser\r\n\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0m\x1b[?25l\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0mu\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0mus\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0muse\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0muser\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0musern\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0muserna\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0musernam\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0musername\x1b[?25h\x1b[0G\x1b[2K\x1b[1;92m? \x1b[0m\x1b[1;99mUsername \x1b[0m\x1b[36musername\x1b[0m\r\n\x1b[1;92m? \x1b[0m\x1b[1;99mPassword \x1b[0m\x1b[?25l\x1b7\x1b[999;999f\x1b[6n\x1b8\x1b[?25h\x1b[6n********\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1D\x1b[1E"`

Comment (makesitgo) [reply internal/commands/login/inputs_test.go:285]: gross

**Pull request merged**